libraries/abstractions/common_io/include/iot_sdio.h

```diff
@@ -0,0 +1,720 @@
+ *   - IOT_SDIO_TRANSFER_FAIL if failed to send command to for receive
+ *                            response from card.
+ */
+int32_t iot_sdio_io_read_extended(IotSdioSlotHandle_t const pxSdioHandle,
+                                   IotSdioFunction_t eFunc,
+                                   bool bBlockMode,
+                                   bool bOpcode,
+                                   uint32_t ulRegAddr,
+                                   uint8_t *pucBuf,
+                                   uint32_t ulCount);
+
+/*!
+ * @brief   iot_sdio_check_card_presence_status() is used to check is a sdio card is present.
+ *
+ * @param[in]   pxSdioHandle         Handle to sdio card slot instance returned in
+ *                                   iot_sdio_open().
+ * @param[out]  bpCardPresenceStatus True if card is present.
+ *                                   False is card is not present.
+ *
+ * @return
+ *   - IOT_SDIO_SUCCESS on successful check.
+ *   - IOT_SDIO_INVALID_VALUE if any parameter is invalid.
+ */
+int32_t iot_sdio_check_card_presence_status(IotSdioSlotHandle_t const pxSdioHandle,
+                                      bool *bpCardPresenceStatus);
+
+/*!
+ * @brief   iot_sdio_set_io_function_callback() is used to set IOFunctionCallback for each I/O function the
+ *          card supports.
+ *
+ *          IOFunctionCallback for an card I/O function is invoked when the corresponding I/O function
+ *          pending interrupt bit is set in the Card's CCCR. Since the I/O function number is parameter to
+ *          IOFunctionCallback, user can also implement one single IOFunctionCallback and set it for all
+ *          I/O functions. Inside the IOFunctionCallback, different action can be taken according the I/O
+ *          function number.
+ *
+ *          The job of a IOFunctionCallback is to service the interrupt condition caused by the
+ *          corresponding I/O function. It should read/write to the function unique area on
+ *          the card to clear pending interrupt. The specific implementation for each
+ *          IOFunctionCallback should be according to the specification of SDIO function it is
+ *          written for.
+ *
+ * @param[in]   pxSdioHandle                     Handle to sdio card slot instance returned in
+ *                                               iot_sdio_open().
+ * @param[in]   eFunc                            IO function number range from eSdioFunction0
+ *                                               to eSdioFunction7.
+ * @param[in]   xIOFunctionCallback              IOFunctionCallback to be set for I/O function eFunc.
+ * @param[in]   pvIOFunctionCallbackUserContext  User context to be passed when IOFunctionCallback is
+ *                                               invoked.
+ *
+ * @return
+ *   - IOT_SDIO_SUCCESS on success.
+ *   - IOT_SDIO_INVALID_VALUE if any parameter is invalid.
+ */
+int32_t iot_sdio_set_io_function_callback(IotSdioSlotHandle_t const pxSdioHandle,
```

Comment:
What are some use cases that applications do in side this IotSdioIOFunctionCallback_t callback?

Reply:
In this callback, each I/O function specific application can perform I/O function specific handling of transfer completed IRQ.

Reply:
I see.